runtime/codert_vm/cnathelp.cpp

```diff
@@ -1605,6 +1605,23 @@ old_fast_jitLookupInterfaceMethod(J9VMThread *currentThread)
 	return slowPath;
 }
 
+void J9FASTCALL
+old_fast_jitLookupDynamicInterfaceMethod(J9VMThread *currentThread)
+{
+	OLD_JIT_HELPER_PROLOGUE(3);
+	DECLARE_JIT_CLASS_PARM(receiverClass, 1);
+	DECLARE_JIT_CLASS_PARM(interfaceClass, 2);
+	DECLARE_JIT_PARM(UDATA, iTableIndex, 3);
+	UDATA iTableOffset = sizeof (struct J9ITable) + iTableIndex * sizeof (UDATA);
+	UDATA vTableOffset = convertITableOffsetToVTableOffset(currentThread, receiverClass, interfaceClass, iTableOffset);
+	Assert_CodertVM_false(0 == vTableOffset);
+	J9Method* method = *(J9Method**)((UDATA)receiverClass + vTableOffset);
+	if (J9_ARE_NO_BITS_SET(J9_ROM_METHOD_FROM_RAM_METHOD(method)->modifiers, J9AccPublic)) {
+		vTableOffset |= J9_DYNAMIC_INTERFACE_VTABLE_OFFSET_NOT_PUBLIC;
```

Comment:
Why not break this into fast/slow path and throw the exception from the C code? It looks to me like you're generating code for a case which never happens (and if it does, the performance of that case is totally irrelevant).

Reply:
Non-public callees are fine for `linkToVirtual()`, so to throw from the helper I would have to either pass an extra boolean parameter to indicate whether non-public methods are an error, or I'd have to split this into two helpers (one allowing non-public methods, and the other disallowing them)

I could take either of those approaches if you much prefer, though I'm not particularly concerned at the moment about the compiler complexity or memory usage caused by this strategy of generating code to test or clear a bit in this case.

@0xdaryl, care to weigh in?

Reply:
I would break it into two helpers and pass the boolean to common code in the C code (as is done in many other instances in this file). Generating code for a case that will only ever execute in specific negative test cases seems unproductive to me.